tests/core/providers/test_async_http_provider.py

```diff
@@ -85,3 +85,22 @@ async def test_user_provided_session() -> None:
     cached_session = await provider.cache_async_session(session)
     assert len(request._async_session_cache) == 1
     assert cached_session == session
+
+async def test_sign_typed_data(self, web3):
```

Comment:
This test should be in `web3/_utils/module_testing/eth_module.py`, and be pretty much a wholesale copy of the tests [here](https://github.com/ethereum/web3.py/blob/master/web3/_utils/module_testing/eth_module.py#L1918-L2023) with the appropriate `async` and `await` syntax.